### PR TITLE
Expose query retry policy via query event listener

### PR DESCRIPTION
**Pull request by losipiuk**
## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# SPI
* Add information query retry policy to `QueryCompletedEvent` and `QueryCreatedEvent`. ({issue}`12898`)
```

### CODE REVIEW

**Review by linzebing — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/event/QueryMonitor.java [hunk 1]
```diff
@@ -140,7 +141,10 @@ public void queryCreatedEvent(BasicQueryInfo queryInfo)
         eventListenerManager.queryCreated(
                 new QueryCreatedEvent(
                         queryInfo.getQueryStats().getCreateTime().toDate().toInstant(),
-                        createQueryContext(queryInfo.getSession(), queryInfo.getResourceGroupId(), queryInfo.getQueryType()),
+                        createQueryContext(queryInfo.getSession(),
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/event/QueryMonitor.java:144]: wrap next line

File: core/trino-main/src/main/java/io/trino/event/QueryMonitor.java [hunk 2]
```diff
@@ -208,7 +212,10 @@ public void queryImmediateFailureEvent(BasicQueryInfo queryInfo, ExecutionFailur
                         ImmutableList.of(),
                         ImmutableList.of(),
                         Optional.empty()),
-                createQueryContext(queryInfo.getSession(), queryInfo.getResourceGroupId(), queryInfo.getQueryType()),
+                createQueryContext(queryInfo.getSession(),
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/event/QueryMonitor.java:215]: wrap next line

File: core/trino-main/src/main/java/io/trino/event/QueryMonitor.java [hunk 3]
```diff
@@ -226,7 +233,10 @@ public void queryCompletedEvent(QueryInfo queryInfo)
                 new QueryCompletedEvent(
                         createQueryMetadata(queryInfo),
                         createQueryStatistics(queryInfo),
-                        createQueryContext(queryInfo.getSession(), queryInfo.getResourceGroupId(), queryInfo.getQueryType()),
+                        createQueryContext(queryInfo.getSession(),
```
Comment (findepi) [inline core/trino-main/src/main/java/io/trino/event/QueryMonitor.java:236]: wrap next line

**Review by findepi — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/event/QueryMonitor.java (hunk 1 above)
Comment (losipiuk) [reply core/trino-main/src/main/java/io/trino/event/QueryMonitor.java:144]: Yeah - I wonder if you can teach IntelliJ to do that

**Pull request merged**